lib/util/itinerary.js

```diff
@@ -74,14 +76,29 @@ function getFirstTransitLeg (itinerary) {
  * (*) For transit itineraries, the first transit leg is used to make
  * the determination. Otherwise, the itinerary startTime is used.
  */
-export function getItineraryDefaultMonitoredDays (itinerary) {
+export function getItineraryDefaultMonitoredDays(itinerary) {
   const firstTransitLeg = getFirstTransitLeg(itinerary)
   const startMoment = firstTransitLeg
     ? moment(firstTransitLeg.serviceDate, 'YYYYMMDD')
     : moment(itinerary.startTime)
   const dayOfWeek = startMoment.day()
 
-  return (dayOfWeek === 0 || dayOfWeek === 6)
-    ? WEEKEND_DAYS
-    : WEEKDAYS
+  return dayOfWeek === 0 || dayOfWeek === 6 ? WEEKEND_DAYS : WEEKDAYS
+}
+
+function legLocationsAreEqual(legLocation, other) {
+  return legLocation.lat === other.lat && legLocation.lon === other.lon
+}
+export function itinerariesAreEqual(itinerary, other) {
+  return (
+    itinerary.legs.length === other.legs.length &&
+    itinerary.legs.every((leg, index) => {
+      const otherLeg = other?.legs?.[index]
+      return (
+        otherLeg.mode === leg.mode &&
+        legLocationsAreEqual(otherLeg?.to, leg?.to) &&
```

Comment:
If `otherLeg` or `leg` is null, does `legLocationsAreEqual` still get called but with a null object? Because it looks like that could cause a crash when it tries to access lat and lon on the null object. This might just be a gap on my understanding of JS trivia.

Reply:
Yeah this feels like a pop quiz javascript behavior question... My understanding is `legLocationsAreEqual` will receive `undefined` instead of an error being thrown. Which means we do need a null check! Added in 8d0c158e